internal/handlers/pg/msg_find.go

```diff
@@ -154,3 +169,67 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 
 	return &reply, nil
 }
+
+// getMaxTimeMSParameter gets maxTimeMS optional parameter.
+func getMaxTimeMSParameter(document *types.Document) (int32, error) {
+	var maxTimeMS int32
+	maxTimeMSParam, err := common.GetOptionalParam(document, "maxTimeMS", maxTimeMS)
+	if err != nil {
+		var commonErr *common.Error
+		if errors.As(err, &commonErr) {
+			return 0, getErrorForInvalidMaxTimeMS(document)
+		}
+
+		return 0, err
+	}
+
+	if maxTimeMSParam < 0 {
+		return 0, common.NewErrorMsg(
+			common.ErrBadValue,
+			fmt.Sprintf("%v value for maxTimeMS is out of range", maxTimeMSParam),
+		)
+	}
+
+	return maxTimeMSParam, nil
+}
+
+// getErrorForInvalidMaxTimeMS returns error for invalid maxTimeMS type.
+func getErrorForInvalidMaxTimeMS(document *types.Document) error {
+	v, err := document.Get("maxTimeMS")
+	if err != nil {
+		return nil
+	}
+
+	switch maxTimeMS := v.(type) {
+	case float64:
+		if math.IsInf(maxTimeMS, -1) {
+			return common.NewErrorMsg(
+				common.ErrBadValue,
+				fmt.Sprintf("%v value for maxTimeMS is out of range", math.MinInt64),
+			)
+		}
+
+		if math.IsInf(maxTimeMS, +1) {
+			return common.NewErrorMsg(
+				common.ErrBadValue,
+				fmt.Sprintf("%v value for maxTimeMS is out of range", math.MaxInt64),
+			)
+		}
+
+		if maxTimeMS > math.MaxInt32 || maxTimeMS < math.MinInt32 {
+			return common.NewErrorMsg(
+				common.ErrBadValue,
+				fmt.Sprintf("%v value for maxTimeMS is out of range", int64(maxTimeMS)),
+			)
+		}
```

Comment:
how about moving the infinite cases to the ".. out of range" case?

Reply:
In such case, our messages will be very different from what mongodb returns.